toolkit/primitives/domain/src/lib.rs

```diff
@@ -751,4 +788,17 @@ mod tests {
 		let from_str = MainchainAddress::from_str(&str).unwrap();
 		assert_eq!(address, from_str);
 	}
+
+	#[test]
+	fn main_chain_address_hash_from_bech32_address() -> Result<(), Box<dyn std::error::Error>> {
+		let bech32_address = "addr_test1vr5vxqpnpl3325cu4zw55tnapjqzzx78pdrnk8k5j7wl72c6y08nd";
+		let expected_hash: [u8; MAINCHAIN_KEY_HASH_LEN] =
+			hex!("e8c300330fe315531ca89d4a2e7d0c80211bc70b473b1ed4979dff2b");
+
+		let hash = super::MainchainKeyHash::from_bech32_address(bech32_address)?;
```

Comment:
for tests it is fine to just unwrap
```suggestion
		let hash = super::MainchainKeyHash::from_bech32_address(bech32_address).unwrap();
```
it provides clear information regarding what happened and where:
`called `Result::unwrap()` on an `Err` value: "invalid checksum"`